flake.nix

```diff
@@ -3,31 +3,49 @@
 
   inputs = {
     nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
-    flake-utils.url = "github:numtide/flake-utils";
-    rust-overlay.url = "github:oxalica/rust-overlay";
-    naersk.url = "github:nmattia/naersk";
+    nixCargoIntegration = {
+      url = "github:yusdacra/nix-cargo-integration";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    flakeCompat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
   };
 
-  outputs = inputs@{ self, nixpkgs, naersk, rust-overlay, flake-utils, ... }:
-    flake-utils.lib.eachDefaultSystem (system:
-      let
-        pkgs = import nixpkgs { inherit system; overlays = [ rust-overlay.overlay ]; };
-        rust = (pkgs.rustChannelOf {
-            date = "2021-05-01";
-            channel = "nightly";
-        }).minimal; # cargo, rustc and rust-std
-        naerskLib = naersk.lib."${system}".override {
-          # naersk can't build with stable?!
-          # inherit (pkgs.rust-bin.stable.latest) rustc cargo;
-          rustc = rust;
-          cargo = rust;
-        };
-      in rec {
-        packages.helix = naerskLib.buildPackage {
-          pname = "helix";
-          root = ./.;
+  outputs = inputs@{ self, nixpkgs, nixCargoIntegration, ... }:
+    let
+      inherit (nixpkgs) lib;
+
+      outputs = nixCargoIntegration.lib.makeOutputs {
+        root = ./.;
+        buildPlatform = "crate2nix";
+        overrides = {
+          crateOverrides = common: prev: {
+            helix-term = old: {
+              buildInputs = (old.buildInputs or [ ]) ++ [ common.cCompiler.cc.lib ];
+            };
+          };
+          shell = common: prev: {
+            packages = prev.packages ++ (with common.pkgs; [ lld_10 lldb ]);
+            env = prev.env ++ [
+              { name = "HELIX_RUNTIME"; eval = "$PWD/runtime"; }
+              { name = "RUST_BACKTRACE"; value = "1"; }
+              { name = "RUSTFLAGS"; value = "-C link-arg=-fuse-ld=lld -C target-cpu=native"; }
```

Comment:
Do we need this? I thought nix defaults no normal optimization instead of native?

Reply:
That was in the `shell.nix` file, so I kept it. Those will only be exported while in the development shell and not when building (when it works that is).